pkg/kgo/compression.go

```diff
@@ -12,8 +12,12 @@ import (
 	"github.com/klauspost/compress/s2"
 	"github.com/klauspost/compress/zstd"
 	"github.com/pierrec/lz4/v4"
+	"github.com/twmb/franz-go/pkg/kgo/internal/pool"
 )
 
+const maxPoolDecodedBufferSize = 8<<20 // 8MB
```

Comment:
can this be a bit larger? 😅 In mimir we'd go as high as fetching 650MiB of decompressed bytes when replaying old data

The config of the client has max fetch bytes config (`FetchMaxBytes`). Can we use that with some compression factor to decide the maximum decompressed batch size?

Reply:
_addressed @ 8d36414e993ed377ebd0b897f7d441d962df5a85_